### ISSUE TITLE
Make addValidation as Protected method

### ISSUE THREAD

**Issue by afranzi** [scallop/scallop#130]
It will be great to define `addValidation` method from `ScallopConf` as Protected, so we can add validation methods in our ScallopConf subclasses as ValidateDateFormat.

```
  private[scallop] def addValidation(fn: => Either[String, Unit]) {
    validations :+= (() => fn)
  }
```

```
import org.joda.time.DateTime
import org.joda.time.format.DateTimeFormatter
import org.rogach.scallop._

class Args(arguments: Seq[String]) extends ScallopConf(arguments) {
  val dtAnalysis: ScallopOption[String] = opt[String]("dt_analysis", 'd', "Target analysis date", required = true)

  validateDateTime(dtAnalysis, DateTimeFormat.forPattern("Y-MM-dd"))
  verify()

  protected def validateDateTime(date: ScallopOption[String], format: DateTimeFormatter): Unit = addValidation {
    date.toOption.map(date => {
      try {
        DateTime.parse(date, format)
        Right(())
      } catch {
        case _: IllegalArgumentException => Left(s"$date must follow $format format")
      }
    }).getOrElse(Right(()))
  }
}
```

**Comment by Rogach** [comment]
Ok, I'll do that. In the meanwhile, you can simply use `validateOpt`:

```scala
protected def validateDateTime(date: ScallopOption[String], format: DateTimeFormatter): Unit =
  validateOpt(date)(dateOpt => {
    dateOpt.map(date => {
      try {
        DateTime.parse(date, format)
        Right(())
      } catch {
        case _: IllegalArgumentException => Left(s"$date must follow $format format")
      }
    }).getOrElse(Right())
  })

```

**Comment by afranzi** [comment]
Cool! Thanks @Rogach!

**Comment by Rogach** [comment]
Released `2.1.0` with this change, should sync to Maven Central within an hour.
